### PR TITLE
refactor: seat tracker snapshot

### PR DESCRIPTION
**Pull request by cristinabaldor**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

union historical snapshot to new dbt snapshot data and standardize historical data to include valid_from and valid_to dates, then union to a date spine of Fridays since 8/4/2023.

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208080148341872

### CODE REVIEW

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+        ) -%}
+
+        {{-
+            dbt_utils.date_spine(
+                start_date="cast('2023-08-04' as date)",
+                end_date=end_date,
+                datepart="week",
+            )
+        -}}
+    )
```
Comment (cristinabaldor) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql:15]: hey @cbini, not sure why i'm getting an error here but I can't get trunk to be happy

Comment (cbini) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql:15]: I think `date_spine` returns a CTE. does this run in dev?

Anyway, use [generate_date_array](https://cloud.google.com/bigquery/docs/reference/standard-sql/array_functions#generate_date_array)

Comment (cristinabaldor) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql:15]: is there a way to set the end date as "today"?

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql [hunk 2]
```diff
@@ -1,17 +1,32 @@
+with
+    date_array as (
+        select
+            generate_date_array(
+                date '2023-08-04',
```
[the remark this answers is not in the document]
Comment (cristinabaldor) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql]: it's just the first friday i have in the archive file, nothing would match to dates before then

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql [hunk 3]
```diff
@@ -1,17 +1,34 @@
+with
+    date_spine as (
+        select date_week,
+        from
+            unnest(
+                generate_date_array(
+                    /* first date of the appsheet snapshot*/
+                    date('2023-08-04'),
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__seat_tracker_snapshot.sql:8]: doesn't need the `date` function

File: src/dbt/kipptaf/models/google/appsheet/intermediate/int_seat_tracker_snapshot.sql
Comment (cbini) [inline src/dbt/kipptaf/models/google/appsheet/intermediate/int_seat_tracker_snapshot.sql:1]: rename to `int_seat_tracker__snapshot.sql`

**Review by cbini — CHANGES_REQUESTED** [review]

**Review by cbini — APPROVED** [review]

**Pull request merged**